src/utils/createTabNavigator.js

```diff
@@ -70,6 +72,30 @@ export default function createTabNavigator(TabView: React.ComponentType<*>) {
       return route.routeName;
     };
 
+    _getAccessibilityLabelText = route => {
+      const { descriptors } = this.props;
+      const descriptor = descriptors[route.key];
+      const options = descriptor.options;
+
+      if (options.tabBarAccessibilityLabel) {
+        return options.tabBarAccessibilityLabel;
+      }
+
+      return this._getLabelText({ route });
+    };
+
+    _getTestID = route => {
+      const { descriptors } = this.props;
+      const descriptor = descriptors[route.key];
+      const options = descriptor.options;
+
+      if (options.tabBarTestID) {
+        return options.tabBarTestID;
+      }
+
+      return route.routeName;
```

Comment:
I think it's better not to set a `testID` if the user hasn't done it explicitly since it might conflict with other `testID` they might have set in the app. So this should be enough:

```js
const { descriptors } = this.props;
const descriptor = descriptors[route.key];

return descriptor.options.tabBarTestID;
```